runners/core-java/src/main/java/org/apache/beam/runners/core/metrics/MonitoringInfoMetricName.java

```diff
@@ -48,6 +45,11 @@ private MonitoringInfoMetricName(String urn, Map<String, String> labels) {
     // and ensure all necessary labels are set for the specific URN.
     this.urn = urn;
     for (Entry<String, String> entry : labels.entrySet()) {
```

Comment:
this.labels = ImmutableMap.copyOf(labels);

Reply:
Done.